server/routes/refreshrsRoutes.js

```diff
@@ -14,6 +14,24 @@ router.get('/', async (req, res, next) => {
   }
 });
 
+// gets all refreshrs belonging to a teacher
+// gonna come back and add authentication middleware later
+router.get('/teachers/:teacherId', async (req, res, next) => {
+  try {
+    const { teacherId } = req.params;
+    const refreshrList = await db.getTeacherRefreshrs(teacherId);
+    console.log(refreshrList);
+    res.status(responseStatus.success).json(refreshrList);
+  } catch (err) {
+    console.log(err);
+    if (TypeError) {
+      next(responseStatus.notFound);
+    } else {
+      next(err);
```

Comment:
My understanding of `next` is that it signifies the end of a middleware function. What does it do here? What would happen without it?

Reply:
https://expressjs.com/en/guide/error-handling.html
I think with an async request, you need to use next to make sure Express processes the error.

Reply:
These are good questions that I don't have an answer to. :smile: I was kinda copying the catches in the other endpoints. I think next would pass it to the error handler? We can refine it later, for sure.

Reply:
Cool, thanks for that link, Kelly!